### ISSUE TITLE
An squeezed empty DataFrame remains a DataFrame

### ISSUE THREAD

**Issue by gbakalian** [pydata/pandas#10436]
not_empty = pd.DataFrame([[None,None]],columns=['a', 'b']).squeeze()
type(not_empty)

empty = pd.DataFrame(columns=['a', 'b']).squeeze()
type(empty)

**Comment by bashtage** [comment]
I don't believe you can make an empty series, so squeezing an empty DataFrame would necessitate losing information about the columns or index since this should produce a 0-dimensional array (not a 1-dimensional array).  

For example, 

```
pd.Series(index=['a','b'])
Out[10]: 
a   NaN
b   NaN
dtype: float64
```

has 2 (null) values.
